src/Pages/Applications/DevtronApps/service.types.ts

```diff
@@ -27,3 +27,30 @@ export interface AppConfigStatusResponseItem {
 }
 
 export type AppConfigStatusResponse = ResponseType<AppConfigStatusResponseItem[]>
+
+export enum ResourceConfigState {
+    Unnamed,
+    Draft,
+    ApprovalPending,
+    Published,
+}
+
+export enum ResourceType {
+    ConfigMap = 'ConfigMap',
+    Secret = 'Secret',
+    DeploymentTemplate = 'Deployment Template',
+}
+
+export interface ResourceConfig {
```

Comment:
This ResourceType conflicts with the global ResourceKindType. I think we can use ConfigResourceType and stick to the convention for all 3